service/src/main/java/uk/gov/hmcts/reform/fpl/model/CaseData.java

```diff
@@ -607,6 +607,33 @@ public boolean hasFutureHearing(List<Element<HearingBooking>> hearingBookings) {
         return defaultIfNull(draftUploadedCMOs, new ArrayList<>());
     }
 
+    public Optional<Element<HearingOrder>> getDraftUploadedCMOWithId(UUID orderId) {
+        return getDraftUploadedCMOs().stream()
+            .filter(draftCmoElement -> orderId.equals(draftCmoElement.getId()))
+            .findFirst();
+    }
+
+    @JsonIgnore
+    public List<Element<HearingOrder>> getHearingOrderDraftCMOs() {
+        if (hearingOrdersBundlesDrafts != null) {
+            return hearingOrdersBundlesDrafts.stream()
+                .map(Element::getValue)
+                .flatMap((HearingOrdersBundle hearingOrdersBundle)
+                    -> hearingOrdersBundle.getCaseManagementOrders().stream())
+                .collect(toList());
+        }
+
+        return new ArrayList<>();
+    }
+
+    public Optional<Element<HearingOrdersBundle>> getHearingOrderBundleThatContainsOrder(UUID orderId) {
+        return hearingOrdersBundlesDrafts.stream()
+            .filter(hearingOrdersBundleElement
+                -> hearingOrdersBundleElement.getValue().getCaseManagementOrders().stream()
+                .anyMatch(o -> o.getId().equals(orderId)))
```

Comment:
let's change 'o' to something more descriptive. `hearingOrderElement` for example.